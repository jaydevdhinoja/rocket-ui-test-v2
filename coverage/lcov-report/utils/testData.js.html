<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils/testData.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">utils</a> testData.js
    </h1>
    <div class='clearfix'>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { LaunchCollectionContext } from '../services/LaunchCollectionProvider';
import { render } from '@testing-library/react';
&nbsp;
export const InitialState = {
    launch: {
        launch_id:1,
        flight_number: 1,
        mission_name: "FalconSat",
        mission_id: [],
        upcoming: false,
        launch_year: "2006",
        launch_date_unix: 1143239400,
        launch_date_utc: "2006-03-24T22:30:00.000Z",
        launch_date_local: "2006-03-25T10:30:00+12:00",
        is_tentative: false,
        tentative_max_precision: "hour",
        tbd: false,
        launch_window: 0,
        rocket: {rocket_id: "falcon1", rocket_name: "Falcon 1", rocket_type: "Merlin A"},
        ships: [],
        telemetry: {flight_club: null},
        launch_site: {site_id: "kwajalein_atoll", site_name: "Kwajalein Atoll", site_name_long: "Kwajalein Atoll Omelek Island"},
        launch_success: false,
        launch_failure_details: {time: 33, altitude: null, reason: "merlin engine failure"},
        links: {mission_patch: "https://images2.imgbox.com/40/e3/GypSkayF_o.png", mission_patch_small: "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png", reddit_campaign: null, reddit_launch: null, reddit_recovery: null},
        details: "Engine failure at 33 seconds and loss of vehicle",
        static_fire_date_utc: "2006-03-17T00:00:00.000Z",
        static_fire_date_unix: 1142553600,
        timeline: {webcast_liftoff: 54}
    },
	launchCollection: [
		{
            launch_id:1,
			flight_number: 1,
			mission_name: "FalconSat",
			mission_id: [],
			upcoming: false,
			launch_year: "2006",
			launch_date_unix: 1143239400,
			launch_date_utc: "2006-03-24T22:30:00.000Z",
			launch_date_local: "2006-03-25T10:30:00+12:00",
			is_tentative: false,
			tentative_max_precision: "hour",
			tbd: false,
			launch_window: 0,
			rocket: {rocket_id: "falcon1", rocket_name: "Falcon 1", rocket_type: "Merlin A"},
			ships: [],
			telemetry: {flight_club: null},
			launch_site: {site_id: "kwajalein_atoll", site_name: "Kwajalein Atoll", site_name_long: "Kwajalein Atoll Omelek Island"},
			launch_success: false,
			launch_failure_details: {time: 33, altitude: null, reason: "merlin engine failure"},
			links: {mission_patch: "https://images2.imgbox.com/40/e3/GypSkayF_o.png", mission_patch_small: "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png", reddit_campaign: null, reddit_launch: null, reddit_recovery: null},
			details: "Engine failure at 33 seconds and loss of vehicle",
			static_fire_date_utc: "2006-03-17T00:00:00.000Z",
			static_fire_date_unix: 1142553600,
			timeline: {webcast_liftoff: 54}
		}
	],
	loading: false,
	current_flight_number: 1,
    selectedRocketId: 'falcon1',
    rocketsData: [
        {
            id: 1,
            active: false,
            stages: 2,
            boosters: 0,
            cost_per_launch: 6700000,
            success_rate_pct: 40,
            first_flight: "2006-03-24",
            country: "Republic of the Marshall Islands",
            company: "SpaceX",
            height: {meters: 22.25, feet: 73},
            diameter: {meters: 1.68, feet: 5.5},
            mass: {kg: 30146, lb: 66460},
            first_stage: {reusable: false, engines: 1, fuel_amount_tons: 44.3, burn_time_sec: 169},
            second_stage: {reusable: false, engines: 1, fuel_amount_tons: 3.38, burn_time_sec: 378},
            engines: {number: 1, type: "merlin", version: "1C", layout: "single"},
            landing_legs: {number: 0, material: null},
            flickr_images: (2) ["https://www.spacex.com/sites/spacex/files/styles/m…f_south_full_wide_ro8a1280_edit.jpg?itok=8loiSGt1", "https://www.spacex.com/sites/spacex/files/styles/m…toff_west_full_wide_nn6p2062_xl.jpg?itok=p776nHsM"],
            wikipedia: "https://en.wikipedia.org/wiki/Falcon_1",
            description: "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.",
            rocket_id: "falcon1",
            rocket_name: "Falcon 1",
            rocket_type: "rocket"
        }
    ],
    setLaunchToggle: jest.fn()
};
&nbsp;
export const renderWithContext = (node, { value, ...options }) =&gt; {
	return render(&lt;LaunchCollectionContext.Provider value={value}&gt;{node}&lt;/LaunchCollectionContext.Provider&gt;, options);
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 02 2020 16:52:00 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
