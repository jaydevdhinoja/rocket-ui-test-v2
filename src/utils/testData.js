import React from 'react'
import { LaunchCollectionContext } from '../services/LaunchCollectionProvider';
import { render } from '@testing-library/react';

export const InitialState = {
    launch: {
        launch_id:1,
        flight_number: 1,
        mission_name: "FalconSat",
        mission_id: [],
        upcoming: false,
        launch_year: "2006",
        launch_date_unix: 1143239400,
        launch_date_utc: "2006-03-24T22:30:00.000Z",
        launch_date_local: "2006-03-25T10:30:00+12:00",
        is_tentative: false,
        tentative_max_precision: "hour",
        tbd: false,
        launch_window: 0,
        rocket: {rocket_id: "falcon1", rocket_name: "Falcon 1", rocket_type: "Merlin A"},
        ships: [],
        telemetry: {flight_club: null},
        launch_site: {site_id: "kwajalein_atoll", site_name: "Kwajalein Atoll", site_name_long: "Kwajalein Atoll Omelek Island"},
        launch_success: false,
        launch_failure_details: {time: 33, altitude: null, reason: "merlin engine failure"},
        links: {mission_patch: "https://images2.imgbox.com/40/e3/GypSkayF_o.png", mission_patch_small: "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png", reddit_campaign: null, reddit_launch: null, reddit_recovery: null},
        details: "Engine failure at 33 seconds and loss of vehicle",
        static_fire_date_utc: "2006-03-17T00:00:00.000Z",
        static_fire_date_unix: 1142553600,
        timeline: {webcast_liftoff: 54}
    },
	launchCollection: [
		{
            launch_id:1,
			flight_number: 1,
			mission_name: "FalconSat",
			mission_id: [],
			upcoming: false,
			launch_year: "2006",
			launch_date_unix: 1143239400,
			launch_date_utc: "2006-03-24T22:30:00.000Z",
			launch_date_local: "2006-03-25T10:30:00+12:00",
			is_tentative: false,
			tentative_max_precision: "hour",
			tbd: false,
			launch_window: 0,
			rocket: {rocket_id: "falcon1", rocket_name: "Falcon 1", rocket_type: "Merlin A"},
			ships: [],
			telemetry: {flight_club: null},
			launch_site: {site_id: "kwajalein_atoll", site_name: "Kwajalein Atoll", site_name_long: "Kwajalein Atoll Omelek Island"},
			launch_success: false,
			launch_failure_details: {time: 33, altitude: null, reason: "merlin engine failure"},
			links: {mission_patch: "https://images2.imgbox.com/40/e3/GypSkayF_o.png", mission_patch_small: "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png", reddit_campaign: null, reddit_launch: null, reddit_recovery: null},
			details: "Engine failure at 33 seconds and loss of vehicle",
			static_fire_date_utc: "2006-03-17T00:00:00.000Z",
			static_fire_date_unix: 1142553600,
			timeline: {webcast_liftoff: 54}
		}
	],
	loading: false,
	current_flight_number: 1,
    selectedRocketId: 'falcon1',
    rocketsData: [
        {
            id: 1,
            active: false,
            stages: 2,
            boosters: 0,
            cost_per_launch: 6700000,
            success_rate_pct: 40,
            first_flight: "2006-03-24",
            country: "Republic of the Marshall Islands",
            company: "SpaceX",
            height: {meters: 22.25, feet: 73},
            diameter: {meters: 1.68, feet: 5.5},
            mass: {kg: 30146, lb: 66460},
            first_stage: {reusable: false, engines: 1, fuel_amount_tons: 44.3, burn_time_sec: 169},
            second_stage: {reusable: false, engines: 1, fuel_amount_tons: 3.38, burn_time_sec: 378},
            engines: {number: 1, type: "merlin", version: "1C", layout: "single"},
            landing_legs: {number: 0, material: null},
            flickr_images: (2) ["https://www.spacex.com/sites/spacex/files/styles/m…f_south_full_wide_ro8a1280_edit.jpg?itok=8loiSGt1", "https://www.spacex.com/sites/spacex/files/styles/m…toff_west_full_wide_nn6p2062_xl.jpg?itok=p776nHsM"],
            wikipedia: "https://en.wikipedia.org/wiki/Falcon_1",
            description: "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.",
            rocket_id: "falcon1",
            rocket_name: "Falcon 1",
            rocket_type: "rocket"
        }
    ],
    setLaunchToggle: jest.fn()
};

export const renderWithContext = (node, { value, ...options }) => {
	return render(<LaunchCollectionContext.Provider value={value}>{node}</LaunchCollectionContext.Provider>, options);
}