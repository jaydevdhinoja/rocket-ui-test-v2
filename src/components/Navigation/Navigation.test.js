import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import Navigation from '.'
import { cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { InitialState, renderWithContext } from '../../utils/testData'

afterEach(cleanup);

describe('Launch Collection Listing', () => {
	test('renders component', () => {
		const { container } = renderWithContext(<MemoryRouter><Navigation/></MemoryRouter>, { value: InitialState });
		expect(container).toMatchSnapshot();
	});
});