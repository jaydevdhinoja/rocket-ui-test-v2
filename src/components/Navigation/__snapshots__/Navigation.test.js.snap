// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Collection Listing renders component 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/launches"
      >
        Launches
      </a>
    </li>
  </ul>
</div>
`;
